scripts/pre_commit_linter.py

```diff
@@ -249,6 +270,8 @@ def _pre_commit_linter():
             all_files = [input_path]
         else:
             all_files = _get_all_files_in_directory(input_path)
+    elif file_list:
+        all_files = file_list
```

Comment:
I think this could do with a little refactoring -- we branch twice on input_path/file_list and I don't think we really need to do that.

Can we move lines 250 - 266 to the very top of the function? Then, branch as follows:

```
   if parsed_args.path:
       # do stuff, compute all_files
   elif parsed_args.files:
       # do stuff, compute all_files
   else:
       # do stuff, compute all_files
```

I think that would be cleaner, what do you think?